src/syntax/syntax.rkt

```diff
@@ -508,8 +509,35 @@
        (define vars (build-list (length ireprs) (lambda (i) (string->symbol (format "x~a" i)))))
        (synth-fl-impl name vars `(,new-op ,@vars))]))
 
+  ; make hash table
+  (define rules (make-hasheq))
+  (define commutes? #f)
+  (when identities
+    (print "looping")
+    (for ([ident (in-list identities)])
+      (match ident
+        [(list ident-name lhs-expr rhs-expr)
+         (cond
+           [(hash-has-key? rules ident-name)
+            (raise-herbie-syntax-error "Duplicate identity ~a" ident-name)]
+           [else (hash-set! rules ident-name (list lhs-expr rhs-expr))])]
+        [(list 'exact expr)
+         (hash-set! rules (gensym (string->symbol (format "~a-exact" name))) (list expr expr))]
+        [(list 'commutes)
+         (cond
+           [commutes? (raise-herbie-syntax-error "Commutes identity already defined")]
+           [else
+            (cond
+              [(not (equal? actual-arity 2))
+               (raise-herbie-syntax-error "Cannot commute a non 2-ary operator")]
+              [else
+               (set! commutes? #t)
+               (hash-set! rules
+                          (string->symbol (format "~a-commutes" name))
```

Comment:
What if someone defines a rule with that name?

Reply:
I'm guessing that should not be allowed @bksaiki

Reply:
Probably